cluster/src/test/scala/io/snappydata/benchmark/snappy/TPCHPerfComparer.java

```diff
@@ -0,0 +1,151 @@
+      reportPrintStream.println("                        "+new Date()+"                                           ");
+      reportPrintStream.println("--------------------------------------------------------------------------------");
+      Stream<Map<Integer, Double>> averages = sortedPaths.map(path ->
+      {
+        int folderCount = atomicCount.incrementAndGet();
+        Map<Integer, Double> perfMap = new HashMap<Integer, Double>();
+        System.out.println("#"+folderCount + " : " + path.getParent().getParent().getParent());
+        reportPrintStream.println("#"+folderCount + " : " + path.getParent().getParent().getParent());
+        //reportPrintStream.println(atomicCount.incrementAndGet() + " : " + path.getParent().getParent().getParent());
+        try {
+          Files.lines(path).map(line->line.split(",")).forEach(element -> perfMap.put(Integer.parseInt(element[0]), Double.parseDouble(element[1])));
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
+        return perfMap;
+      });
+
+      List<Map<Integer, Double>> averageList = averages.collect(Collectors.toList());
+      System.out.println("---------------------------------------------------------------------------------------");
+      reportPrintStream.println("--------------------------------------------------------------------------------");
+
+      System.out.print("Query");
+      reportPrintStream.print("Query");
+      for(int j = 0 ; j < atomicCount.get(); j++) {
+        System.out.print("     #" + j);
+        reportPrintStream.print("     #" + j);
+      }
+      System.out.println();
+      reportPrintStream.println();
+
+      for(int i=1; i < 23; i++){
+        System.out.print(i < 10 ? " 0" + i : " "+i);
+        reportPrintStream.print(i < 10 ? " 0" + i : " "+i);
+        System.out.print("     ");
+        reportPrintStream.print("     ");
+        int count = 0;
+        Double firstValue = 0.0;
+        int whichRevPerfDown = 0;
+        for (Map<Integer, Double> singleMap : averageList) {
+          if (count == 0) {
+            if(singleMap.get(i) != null) {
+              firstValue = singleMap.get(i);
+              System.out.print("---");
+              reportPrintStream.print("---");
+              //System.out.print(firstValue);
+              count++;
+            }else{
+              System.out.print("        ");
+            }
+          } else {
+            if (singleMap.get(i) != null) {
+              double secondValue = singleMap.get(i);
+              double perf = (firstValue-secondValue)/secondValue;
+              perf = perf > 0 ? 1 + perf : -1 + perf;
+              if(perf > 0){
+                System.out.print("    ");
+                reportPrintStream.print("    ");
+              }else{
+                System.out.print("   ");
+                reportPrintStream.print("   ");
+                if(perf < -1.15){
```

Comment:
Can you make this as 5% for raising the flag for notification of performance degradation? 15 % looks higher.

Reply:
5% falls in noise limits. Maybe something in between like 10%.